src/Server/TextDocument.php

```diff
@@ -148,4 +167,14 @@ public function formatting(TextDocumentIdentifier $textDocument, FormattingOptio
         return [$edit];
     }
     
+    public function completion(TextDocumentIdentifier $textDocument, Position $position)
+    {
+        $completionReporter = new CompletionReporter();
+        $tokens = token_get_all($this->documents[$textDocument->uri]);
```

Comment:
This part was written just to show results, I knew I will need to rebase this PR after #31 merge. 

List of tokens is always available after Parser::parse(), Lexer::getTokens() contains such array and it can be used without executing token_get_all() once again. Even without that dividing into tokens is very fast :) Main problem with pure tokens array is that its a little bit inconsistent and it needs to be converted into something more useful for line:column processing. In this case conversion changes all arrays elements into the same object type with document range. 

Mixing ast with token_get_all is in theory possible because token_get_all can process part of code ('<?php' tag needs to be added) but I'm not sure if its best solution. It can be tricky in my opinion.

Token approach looks right now to be good solution in my opinion but its just my version and maybe someone will came with better implementation :)